src/include/snapper/dialogs.rb

```diff
@@ -47,198 +44,409 @@ def initialize_snapper_dialogs(include_target)
-          "userdata"    => get_userdata("userdata")
+          type:        UI.QueryWidget(Id(:rb_type), :Value),
+          description: UI.QueryWidget(Id("description"), :Value),
+          pre_number:  UI.QueryWidget(Id(:pre_list), :Value),
+          cleanup:     UI.QueryWidget(Id("cleanup"), :Value),
+          user_data:   userdata("userdata")
         }
-        if type == :POST
-          pre_args = {
-            "num"         => pre_num,
-            "description" => UI.QueryWidget(Id("pre_description"), :Value),
-            "cleanup"     => UI.QueryWidget(Id("pre_cleanup"), :Value),
-            "userdata"    => get_userdata("pre_userdata")
-          }
-        end
         break if ret == :ok || ret == :cancel
       end
+
       UI.CloseDialog
-      if ret == :ok
-        if snapshot_modified(snapshot, args)
-          modified = Snapper.ModifySnapshot(args)
+      created = Snapper.CreateSnapshot(args) if ret == :ok
+      created
+    end
+
+    # Popup for deleting existing snapshot
+    # @return true if snapshot was deleted
+    def DeleteSnapshotPopup(snapshots)
+      snaps = []
+      post_snaps = []
+      snapshots.each do |s|
+        if !s.post?
+          snaps << s.number
+        else
+          post_snaps << { pre_num: s.pre.number, num: s.number }
         end
-        if type == :POST && snapshot_modified(pre_snapshot, pre_args)
-          modified = Snapper.ModifySnapshot(pre_args) || modified
+      end
+
+      nums = snaps + post_snaps.map { |k| "#{k[:pre_num]}&#{k[:num]}" }
+
+      # yes/no popup question
+      return false if !Popup.YesNo(_("You have selected those snapshots: %{num}") % { num: nums.join(",") })
+
+      Snapper.DeleteSnapshot(snaps)
+      post_snaps.each do |k|
+        Snapper.DeleteSnapshot([k[:pre_num], k[:num]])
+      end
+      true
+    end
+
+    # Summary dialog
+    # @return dialog result
+    def SummaryDialog
+      # summary dialog caption
+      caption = _("Snapshots")
+
+      # update list of snapshots
+      Wizard.SetContentsButtons(
+        caption,
+        snapshots_table,
+        Ops.get_string(@HELPS, "summary", ""),
+        Label.BackButton,
+        Label.CloseButton
+      )
+      Wizard.HideBackButton
+      Wizard.HideAbortButton
+
+      UI.SetFocus(Id(:snapshots_table))
+
+      refresh_buttons
+      summary_event_loop
+    end
+
+    def format_diff(diff, textmode)
+      lines = String.EscapeTags(diff).split("\n")
+      return lines.join("<br>") if textmode
+      # colorize diff output
+      lines.map! do |line|
+        case line[0]
+        when "+"
+          line = "<font color=blue>#{line}</font>"
+        when "-"
+          line = "<font color=red>#{line}</font>"
         end
+        line
       end
 
-      modified
+      # show fixed font in diff
+      "<pre>#{lines.join("<br>")}</pre>"
     end
 
+    def textmode?
+      display_info = UI.GetDisplayInfo
+      display_info["TextMode"] || false
```

Comment:
the original code was something like ```Ops.get_boolean(display_info["TextMode"], false)``` so it could be considered just a direct replacement